build_projects/shared-build-targets-utils/Utils/Dirs.cs

```diff
@@ -47,9 +47,9 @@ private static string GetNuGetPackagesDir()
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                return Path.Combine(Environment.GetEnvironmentVariable("USERPROFILE"), ".nuget", "packages");
+                return Path.Combine(Dirs.RepoRoot, ".nuget", "packages");
```

Comment:
Not sure what you mean, Dirs.RepoRoot is what makes the cache inside the repo so we don't have build-to-build conflicts.

The method you linked will read from the environment variable that we set here.